packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -0,0 +1,253 @@
+/**
+ * @file Transaction Resolver - handles cross-chain transaction confirmations
+ *
+ * This resolver is responsible for tracking pending transactions and providing
+ * promises/vows that resolve when transactions are confirmed.
+ * This is an orchestration component that can be used independently of portfolio logic.
+ */
+
+import { makeTracer, mustMatch } from '@agoric/internal';
+import type { CaipChainId, AccountId } from '@agoric/orchestration';
+import type { Zone } from '@agoric/zone';
+import { type VowTools, VowShape } from '@agoric/vow';
+import type { ZCF, ZCFSeat } from '@agoric/zoe';
+import { M } from '@endo/patterns';
+import type {
+  CCTPTransactionEntry,
+  CCTPTransactionKey,
+  CCTPResolverServiceFacet,
+  CCTPSettlementOfferArgs,
+} from './types.js';
+import { CCTPTransactionDetailsShape } from './types.js';
+
+const trace = makeTracer('Resolver');
+
+/**
+ * Prepare the CCTP Transaction Resolver.
+ *
+ * This is an orchestration component that can be used independently
+ * of portfolio logic to handle CCTP transaction confirmations.
+ *
+ * @param zone - Durable storage zone
+ * @param vowTools - Vow tools for creating promises
+ * @returns A function to create resolver instances with separate facets
+ */
+export const prepareCCTPResolver = (zone: Zone, vowTools: VowTools) => {
+  const resolverZone = zone.subZone('CCTPResolver');
+  const cctpTransactionRegistry = resolverZone.mapStore<
+    CCTPTransactionKey,
+    CCTPTransactionEntry
+  >('cctpTransactionRegistry');
+
+  const ClientFacetI = M.interface('CCTPResolverClient', {
+    registerCCTPTransaction: M.call(M.string(), M.bigint()).returns(VowShape),
```

Comment:
the amount can't be negative, right?

```suggestion
    registerCCTPTransaction: M.call(M.string(), M.nat()).returns(VowShape),
```